snowplow-common-internal/kinesis/src/main/scala/com/snowplowanalytics/snowplow/sources/kinesis/KinesisSourceConfig.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2023-present Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Snowplow Community License Version 1.0,
+ * and you may not use this file except in compliance with the Snowplow Community License Version 1.0.
+ * You may obtain a copy of the Snowplow Community License Version 1.0 at https://docs.snowplow.io/community-license-1.0
+ */
+package com.snowplowanalytics.snowplow.sources.kinesis
+
+import cats.implicits._
+import eu.timepit.refined.types.all.PosInt
+import io.circe._
+import io.circe.generic.extras.semiauto.deriveConfiguredDecoder
+import io.circe.generic.extras.Configuration
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.regions.providers.DefaultAwsRegionProviderChain
+import software.amazon.kinesis.common.{InitialPositionInStream, InitialPositionInStreamExtended}
+
+import java.net.URI
+import java.time.Instant
+import java.util.Date
+
+case class KinesisSourceConfig(
+  appName: String,
+  streamName: String,
+  initialPosition: InitialPositionInStreamExtended,
+  retrievalMode: KinesisSourceConfig.Retrieval,
+  bufferSize: PosInt,
+  customEndpoint: Option[URI],
+  dynamodbCustomEndpoint: Option[URI],
+  cloudwatchCustomEndpoint: Option[URI]
+)
+
+object KinesisSourceConfig {
+
+  implicit val config: Configuration = Configuration.default.withDefaults
```

Comment:
I need to think about this... whether it's better to define the `Configuration` here, or whether it's better to pass a `Configuration` implicitly when creating the decoder.

I'll come back to this.